### DISCUSSION TITLE
Feature Request - Integrating with Nextcloud

### DISCUSSION THREAD

**Discussion by beffective** [TriliumNext/Trilium#1531] [Ideas]
Hi

I just set up NextCloud and other apps on my home server, as a foundation for my 2021 business.  The intent is also to minimize my app exposure to the big boys like Google, Microsoft, etc.

I am using NextCloud notes and deck features, but the notes part was not enough.  And then I found Trilium, great job by the way. 

Is integrating to Nextcloud on the roadmap, in any way?

**Comment by zadam** [comment]
Hi, unfortunately no integration with nextcloud is planned.

**Reply by 6-AND-9** [reply]
I would second this request. Making an offline Nextcloud app would be a dream come true!

**Comment by sigaloid** [comment]
Maybe mount the Nextcloud Files FTP server locally then save the document.db there.

**Reply by zadam** [reply]
I can't imagine that this would work well. Most probably it would end with a similar result like (frequent) attempts to synchronize using dropbox/google drive - fatal database corruption.

**Reply by sigaloid** [reply]
Wait, I do this, is this something I should not be doing? I have the document.db saved and when I am done working or every once in a while, I copy it over to a network mount as a backup (i also have a sync server though, it's just a backup). Do you mean simultaneous access will corrupt it?

**Reply by 6-AND-9** [reply]
> Wait, I do this, is this something I should not be doing? I have the document.db saved and when I am done working or every once in a while, I copy it over to a network mount as a backup (i also have a sync server though, it's just a backup). Do you mean simultaneous access will corrupt it?

I guess this can fine only if you work on the database from a single point, and you just backup it elsewhere. Otherwise it wouldn't merge changes. I suppose that's what Zadam was trying to convey, correct me if I am wrong.

**Reply by zadam** [reply]
Yes, if you really make sure that the file is fully synced before you access it from some instance then it can work. But it's very easy to make a mistake and cause a corruption. Another common mistake is syncing/backing up the DB file while Trilium is running, that's also a recipe for corruption.

**Comment by chee** [comment]
> Hi
> 
> I just set up NextCloud and other apps on my home server, as a foundation for my 2021 business. The intent is also to minimize my app exposure to the big boys like Google, Microsoft, etc.
> 
> I am using NextCloud notes and deck features, but the notes part was not enough. And then I found Trilium, great job by the way.
> 
> Is integrating to Nextcloud on the roadmap, in any way?

Have you tried setting up the Trilium Server on your server? I have Trilium running on the same server as my NextCloud instance and it works well. What features would you want from NextCloud integration that aren't available from using the Trilium Server?

**Reply by 6-AND-9** [reply]
How do you do this? Through Docker or another way? Can you please explain the process?
